Source/Diagnostics/Diagnostics.cpp

```diff
@@ -84,9 +84,77 @@ Diagnostics::BaseReadParameters ()
     // If user requests to plot proc_number for a serial run,
     // delete proc_number from fields_to_plot
     if (amrex::ParallelDescriptor::NProcs() == 1){
-        m_varnames.erase(
-            std::remove(m_varnames.begin(), m_varnames.end(), "proc_number"),
-            m_varnames.end());
+        m_varnames_fields.erase(
+            std::remove(m_varnames_fields.begin(), m_varnames_fields.end(), "proc_number"),
+            m_varnames_fields.end());
+    }
+
+    // Get names of particle quantities to average at each grid point
+    bool pfield_varnames_specified = pp_diag_name.queryarr("particle_fields_to_plot", m_pfield_varnames);
+    if (!pfield_varnames_specified){
+        m_pfield_varnames = {};
+    }
+#ifdef WARPX_DIM_RZ
+    if (m_pfield_varnames.size() != 0) {
+        amrex::Abort("Input error: cannot use particle_fields_to_plot not implemented for RZ");
+    }
+#endif
+
+
+    // Get parser strings for particle fields and generate map of parsers
+    bool pfield_parser_specified = false;
+    std::string parser_str;
+    amrex::ParmParse pp_diag_pfield(m_diag_name + ".particle_fields");
+    for (const auto& var : m_pfield_varnames) {
+        Store_parserString(pp_diag_pfield, (var + "(x,y,z,ux,uy,uz)").c_str(), parser_str);
+        if (parser_str != "") {
+            m_pfield_strings.insert({var, parser_str});
+        }
+        else {
+            amrex::Abort("Input error: cannot find parser string for " + var + ". " +
+                         m_diag_name + ".particle_fields." + var + " in file");
+        }
+    }
+
+    // Names of all species in the simulation
+    m_all_species_names = warpx.GetPartContainer().GetSpeciesNames();
+
+    // Get names of species to average at each grid point
+    bool pfield_species_specified = pp_diag_name.queryarr("particle_fields_species", m_pfield_species);
+    if (!pfield_species_specified){
+        m_pfield_species = m_all_species_names;
+    }
+
+    // Check that species names specified in m_pfield_species are valid
+    bool p_species_name_is_wrong;
+    // Loop over all species specified above
+    for (const auto& species : m_pfield_species) {
+        // Boolean used to check if species name was misspelled
+        p_species_name_is_wrong = true;
+        // Loop over all species
+        for (int i = 0, n = int(m_all_species_names.size()); i < n; i++) {
+            // Check if species name extracted from the string rho_<species_name>
```

Comment:
Shall we update this comment and the one right below without mentioning `rho_<species_name>`, `RhoFunctor` and `rho`?